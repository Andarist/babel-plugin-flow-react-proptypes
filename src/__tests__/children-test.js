var babel = require('babel-core');
var content = `
const React = require('react');

type FooProps = {
  children: React.Element[],
};

function Foo(props: FooProps) {
    <div>{props.children}</div>
}
`;

it('children', () => {
  var res = babel.transform(content, {
    babelrc: false,
    presets: ['es2015', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
