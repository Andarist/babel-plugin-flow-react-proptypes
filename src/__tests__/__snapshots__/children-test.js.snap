// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`children 1`] = `
"'use strict';

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var React = require('react');
var Element = React.Element;

Foo.propTypes = {
  children: _propTypes2.default.node.isRequired,
  children2: _propTypes2.default.arrayOf(typeof Element === 'function' ? _propTypes2.default.instanceOf(Element) : _propTypes2.default.any).isRequired,
  children3: typeof Element === 'function' ? _propTypes2.default.instanceOf(Element).isRequired : _propTypes2.default.any.isRequired,
  children4: typeof Element === 'function' ? _propTypes2.default.instanceOf(Element).isRequired : _propTypes2.default.any.isRequired
};


function Foo(props) {
  React.createElement(
    'div',
    null,
    props.children
  );
}"
`;
